services/slack/service.go

```diff
@@ -128,15 +168,15 @@ func (s *Service) Test(options interface{}) error {
 	if !ok {
 		return fmt.Errorf("unexpected options type %T", options)
 	}
-	return s.Alert(o.Channel, o.Message, o.Username, o.IconEmoji, o.Level)
+	return s.Alert("", o.Channel, o.Message, o.Username, o.IconEmoji, o.Level)
 }
 
-func (s *Service) Alert(channel, message, username, iconEmoji string, level alert.Level) error {
-	url, post, err := s.preparePost(channel, message, username, iconEmoji, level)
+func (s *Service) Alert(workspace, channel, message, username, iconEmoji string, level alert.Level) error {
+	url, post, err := s.preparePost(workspace, channel, message, username, iconEmoji, level)
 	if err != nil {
 		return err
 	}
-	client := s.clientValue.Load().(*http.Client)
+	client := s.clients[workspace]
```

Comment:
How do you protect access to the clients dict here?

Reply:
It might be worth creating a Workspace struct that contains both a Config and http.Client so you do not have to maintain and protect access to two maps.